config/config_auth_details.go

```diff
@@ -0,0 +1,130 @@
+package config
+
+import (
+	"fmt"
+	"strings"
+
+	"golang.org/x/exp/slices"
+)
+
+// AuthDetails contains the details of the authentication configuration.
+type AuthDetails struct {
+	AuthType      string            `json:"auth_type"`
+	Username      string            `json:"username,omitempty"`
+	Host          string            `json:"host,omitempty"`
+	AccountID     string            `json:"account_id,omitempty"`
+	Configuration AuthConfiguration `json:"configuration"`
+}
+
+func (a *AuthDetails) String() string {
+	s := ""
+	if a.Host != "" {
+		s += fmt.Sprintf(`host=%s`, a.Host)
+	}
+
+	if a.AuthType != "" {
+		s += fmt.Sprintf(", auth_type=%s", a.AuthType)
+	}
+
+	if a.Username != "" {
+		s += fmt.Sprintf(", username=%s", a.Username)
+	}
+
+	if a.AccountID != "" {
+		s += fmt.Sprintf(", account_id=%s", a.AccountID)
+	}
+
+	conf := a.Configuration.String()
+	if conf != "" {
+		s += fmt.Sprintf(`
+Configuration:
+%s`, conf)
+	}
+
+	return s
+}
+
+// AuthConfiguration is a map of attribute name to its configuration.
+type AuthConfiguration map[string]*AttrConfig
+
+func (c *AuthConfiguration) String() string {
+	keys := make([]string, 0, len(*c))
+	for k := range *c {
+		keys = append(keys, k)
+	}
+	slices.Sort(keys)
+
+	var conf []string
+	for _, k := range keys {
+		v := (*c)[k]
+		conf = append(conf, fmt.Sprintf("- %s=%s", k, v))
+	}
+	return strings.Join(conf, "\n")
+}
+
+type AttrConfig struct {
+	Value            string  `json:"value"`
+	Source           *Source `json:"source"`
+	AuthTypeMismatch bool    `json:"auth_type_mismatch"`
+}
+
+func (a *AttrConfig) String() string {
+	s := fmt.Sprintf(`%s (from %s)`, a.Value, a.Source.String())
+	if a.AuthTypeMismatch {
+		s += " (not used by the current auth type)"
+	}
+	return s
+}
+
+func (c *Config) GetAuthDetails(showSensitive bool) AuthDetails {
```

Comment:
Could we default this to never include sensitive values?

A bool argument can be deceptive. If we need a way to include the sensitive fields, I would rather have a separate function call that makes it clear it is including sensitive values verbatim.